shell/platform/android/io/flutter/embedding/engine/FlutterJNI.java

```diff
@@ -106,6 +106,24 @@
   // platform thread and doesn't require locking.
   private ReentrantReadWriteLock shellHolderLock = new ReentrantReadWriteLock();
 
+  // Prefer using the FlutterJNI.Factory so it's easier to test.
+  public FlutterJNI() {
+    // We cache the main looper so that we can ensure calls are made on the main thread
+    // without consistently paying the synchronization cost of getMainLooper().
+    mainLooper = Looper.getMainLooper();
+  }
+
+  /**
+   * A factory for creating {@code FlutterJNI} instances. Useful for FlutterJNI injections during
+   * tests.
+   */
+  public static class Factory {
```

Comment:
Some of the changes to this file are for consistency purposes.

1. Put private methods right above the public ones.
2. Move constructor and factory to the top. (It's weird that it's at the end)
3. Move annotation after the javadoc.

 I think this file still needs some additional clean ups.